src/main/java/ui/TestController.java

```diff
@@ -157,4 +159,30 @@ public static RepoIO createTestingRepoIO(Optional<JSONStore> jsonStoreToBeUsed)
         return new RepoIO(Optional.of(new DummySource()), jsonStoreToBeUsed,
                           Optional.of(RepoStore.TEST_DIRECTORY));
     }
+
+    /**
+     * Create an update progress window if not in test mode.
+     *
+     * @return update progress window
+     */
+    public static UpdateProgressWindow createUpdateProgressWindow() {
+        if (isTestMode()) {
+            return null;
```

Comment:
If I instantiate the window in the method below, `UI` will have no direct reference to the window. I will need a reference to the window in `MenuControl` to show update window. Would it make sense to have `ui.updateManager.showUpdateWindow`?

After changing the caller in `PlatformEx`, all the tests seem to be fine.